ClientLib/src/main/java/org/droidplanner/services/android/impl/core/drone/autopilot/apm/solo/ArduSolo.java

```diff
@@ -257,7 +278,11 @@ public void run() {
     @Override
     public void notifyDroneEvent(final DroneInterfaces.DroneEventsType event) {
         switch (event) {
-            case HEARTBEAT_FIRST:
+            case HEARTBEAT_FIRST: {
```

Comment:
Interesting. So much indirection with the use of extends can make the code hard to follow! It's would almost be better if the GenericMavLinkDrone object received all messages first and then call a potentially derived processOnMavLinkMessageReceived implemented in specializations to help with the synchronization issues we are seeing. (i.e. the checkIfFlying call is obviously not happening on first connection before anything else)